library/alloc/src/string.rs

```diff
@@ -1392,6 +1392,8 @@ impl String {
 
     /// Appends the given [`char`] to the end of this `String`.
     ///
+    /// Note: When pushing a char, it is automatically converted to its UTF-8 byte representation since Rust strings are UTF-8 encoded.
```

Comment:
```suggestion
    ///  Since Rust strings are UTF-8 encoded, the pushed char will automatically converted to its UTF-8 byte representation.
```

Potential alternate phrasing, not sure which is better, will let reviewer decide.